api/src/pcapi/core/educational/utils.py

```diff
@@ -53,6 +56,11 @@ def log_information_for_data_purpose(
         user_id = get_hashed_user_id(user_email)
         extra_data["userId"] = user_id
 
+    if uai is not None:
+        extra_data["uai"] = uai
+
+    extra_data["user_role"] = user_role
```

Comment:
On devrait vérifier que `user_role` n'est pas nul, non ? Comme pour `user_email` ou `uai`.

Reply:
C'est corrigé, par contre je ne comprend pas l'utilité de mettre if ... is not None: 
Parce que au pire si on met extra_data["gnagnagna"] = truc ( et que truc et None) 
et bien... ca devrait annuler tout seul et faire que extra_data["gnagnagna"] soit None... Non ?

Reply:
La différence c'est que tu ajoutes une clé dans le `dict`. Je ne sais pas pourquoi on évite ça, mais il vaut mieux rester cohérent.